src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -50,34 +63,297 @@ export interface CodeActionShowOptions {
 	readonly includeDisabledActions: boolean;
 	readonly fromLightbulb?: boolean;
 }
+export interface ICodeActionMenuItem {
+	title: string;
+	detail: string;
+	action: IAction;
+	decoratorRight?: string;
+	isSeparator?: boolean;
+	isEnabled: boolean;
+	index: number;
+	disposables?: IDisposable[];
+}
+
+export interface ICodeMenuOptions {
+	useCustomDrawn?: boolean;
+	ariaLabel?: string;
+	ariaDescription?: string;
+	minBottomMargin?: number;
+	optionsAsChildren?: boolean;
+}
 
-export class CodeActionMenu extends Disposable {
+export interface ICodeActionMenuTemplateData {
+	root: HTMLElement;
+	text: HTMLElement;
+	detail: HTMLElement;
+	decoratorRight: HTMLElement;
+	disposables: IDisposable[];
+}
 
-	private _visible: boolean = false;
+const TEMPLATE_ID = 'codeActionWidget';
+class CodeMenuRenderer implements IListRenderer<ICodeActionMenuItem, ICodeActionMenuTemplateData> {
+	get templateId(): string { return TEMPLATE_ID; }
+
+	renderTemplate(container: HTMLElement): ICodeActionMenuTemplateData {
+		const data: ICodeActionMenuTemplateData = Object.create(null);
+		data.disposables = [];
+		data.root = container;
+		data.text = document.createElement('span');
+		// data.detail = document.createElement('');
```

Comment:
Make sure to remove these any of this commented code if you're not planning to restore it